torch_frame/gbdt/tuned_xgboost.py

```diff
@@ -225,8 +240,19 @@ def _predict(self, tf_test: TensorFrame) -> Tensor:
         dtest = xgboost.DMatrix(test_feat, label=test_y,
                                 feature_types=test_feat_type,
                                 enable_categorical=True)
-        pred = self.model.predict(dtest)
-        return torch.from_numpy(pred).to(device)
+        if self.model.best_iteration is not None:
+            iteration_range = self.model.best_iteration
+        else:
+            iteration_range = None
+        pred = self.model.predict(dtest, iteration_range)
+
+        if (self.model.best_iteration
+                and self.task_type == TaskType.MULTICLASS_CLASSIFICATION):
+            assert pred.shape[1] == self._num_classes
+            pred = torch.argmax(torch.from_numpy(pred), dim=1)
+        else:
+            pred = torch.from_numpy(pred)
+        return pred.to(device)
```

Comment:
curious - why do we need to introduce these?

Reply:
Ok, now I understand. Do you mind adding a comment?

Reply:
Sure